cajita/src/Cajita_ParameterPack.hpp

```diff
@@ -12,169 +12,45 @@
-    using element_type = ParameterPackElement<N, value_type<N>>;
-
-    static constexpr std::size_t size = sizeof...( Types );
-};
-
-//---------------------------------------------------------------------------//
 // Static type checker.
-template <class>
-struct is_parameter_pack_impl : public std::false_type
+template <class T>
+struct is_parameter_pack : public Cabana::is_parameter_pack_impl<
+                               typename std::remove_cv<T>::type>::type
 {
 };
 
+// Create a parameter pack.
 template <typename... Types>
-struct is_parameter_pack_impl<ParameterPack<Types...>> : public std::true_type
+Cabana::ParameterPack<Types...> makeParameterPack( const Types&... ts )
 {
-};
-
-template <class T>
-struct is_parameter_pack
-    : public is_parameter_pack_impl<typename std::remove_cv<T>::type>::type
-{
-};
+    auto pp = Cabana::makeParameterPack( ts... );
+    return pp;
+}
 
-//---------------------------------------------------------------------------//
 // Get an element from a parameter pack.
 template <std::size_t N, class ParameterPack_t>
 KOKKOS_FORCEINLINE_FUNCTION typename std::enable_if<
-    is_parameter_pack<ParameterPack_t>::value,
+    Cabana::is_parameter_pack<ParameterPack_t>::value,
     typename ParameterPack_t::template value_type<N>&>::type
 get( ParameterPack_t& pp )
 {
-    return static_cast<typename ParameterPack_t::template element_type<N>&>(
-               pp )
-        ._m;
+    return Cabana::get<N, ParameterPack_t>( pp );
 }
 
+// Get an element from a parameter pack.
 template <std::size_t N, class ParameterPack_t>
 KOKKOS_FORCEINLINE_FUNCTION typename std::enable_if<
-    is_parameter_pack<ParameterPack_t>::value,
+    Cabana::is_parameter_pack<ParameterPack_t>::value,
     const typename ParameterPack_t::template value_type<N>&>::type
 get( const ParameterPack_t& pp )
 {
-    return static_cast<
-               const typename ParameterPack_t::template element_type<N>&>( pp )
-        ._m;
-}
-
-//---------------------------------------------------------------------------//
-// Fill a parameter pack. Note the indexing is such that the Nth element of a
-// parameter pack is the Nth element of the tuple.
-template <typename ParameterPack_t, typename T, typename... Types>
-void fillParameterPackImpl( ParameterPack_t& pp,
-                            const std::integral_constant<std::size_t, 0>,
-                            const T& t, const Types&... )
-{
-    get<ParameterPack_t::size - 1>( pp ) = t;
-}
-
-template <typename ParameterPack_t, std::size_t N, typename T,
-          typename... Types>
-void fillParameterPackImpl( ParameterPack_t& pp,
-                            const std::integral_constant<std::size_t, N>,
-                            const T& t, const Types&... ts )
-{
-    get<ParameterPack_t::size - 1 - N>( pp ) = t;
-    fillParameterPackImpl( pp, std::integral_constant<std::size_t, N - 1>(),
-                           ts... );
+    return Cabana::get<N, ParameterPack_t>( pp );
```

Comment:
I think we can probably remove this file altogether. This is a relatively new expert-level capability so I'm fine changing namespaces to avoid confusion.